yaks/README.md

```diff
@@ -665,11 +678,42 @@ yaks.call(array_of_widgets, mapper: MyCollectionMapper, item_mapper: WidgetMappe
 ```
 
 If the mapper is left unspecified, Yaks will inspect whatever you pass
-it, and try various constant lookups based on naming. These all happen
+it. First it will test the given object against the mapping defined using `mapper_for`.
+If no mapper is found, it wil call `derive_mapper_from_item` or `derive_mapper_from_collection`
+depending on whether the given object is a collection or not. If the object responds
+to `to_ary` it is considered a collection.
+
+Notes:
+* you may override one of these two methods independently from the other.
+* this function takes either a symbol or any instance
```

Comment:
I think you don't need these notes, to keep the README shorter. You can remove the first one after adding description of hierarchy between `derive_mapper_from_*` methods. And the second one is just enumerating what types of object the `derive_mapper_from_*` methods are accepting. I think it would be better to add the argument types in the comments above the methods in the markdown code block above.